### PR TITLE
complete task 5 get-api-articles

### PR DESCRIPTION
**Pull request by Ben04051**
complete task 5

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -41,10 +41,28 @@ describe("GET/api", () => {
...
+           articles.forEach((article) => {
+           expect(typeof article.author).toBe("string")
+           expect(typeof article.title).toBe("string")
+           expect(typeof article.article_id).toBe("number")
+           expect(typeof article.topic).toBe("string")
+           expect(typeof article.created_at).toBe("string")
+           expect(typeof article.votes).toBe("number")
+           expect(typeof article.article_img_url).toBe("string")
+           expect(typeof article.comment_count).toBe("number")
+           expect(Object.keys(article).length).toBe(8)
```
Comment (nc-poonam-rajput) [inline __tests__/app.test.js:60]: again you're making your testing brittle by asserting there can only be 8 keys. You've already asserted you need the above properties to be accounted for so this line isn't needed

File: __tests__/app.test.js [hunk 2]
```diff
@@ -41,10 +41,28 @@ describe("GET/api", () => {
...
-
-
+describe("GET/api/articles", () => {
+    test("200: gets all articles, with the body property removed, a commment count added and sorted by date in descending order", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+           const {articles} = body
+           articles.forEach((article) => {
```
Comment (nc-poonam-rajput) [inline __tests__/app.test.js:51]: You're using a forEach which should immediately be a flag that you need to check this isn't a false positive and a empty array has been returned back. Remember to check this array actually has the right amount of articles inside it

File: __tests__/app.test.js [hunk 3]
```diff
@@ -41,10 +41,28 @@ describe("GET/api", () => {
     });
 })
 
-
-
-
-
+describe("GET/api/articles", () => {
+    test("200: gets all articles, with the body property removed, a commment count added and sorted by date in descending order", () => {
```
Comment (nc-poonam-rajput) [inline __tests__/app.test.js:45]: You've mentioned that the body property should be removed, can you assert in the test that this is the case

Note: this test block is doing a LOT of work. in TDD we really want to contain behaviours in their own tests. Really this looks like four tests:
1. gets all articles
2. body property isn't there
3. comment count is there
4. it is sorted by date in desc order default

**Review by nc-poonam-rajput — COMMENTED** [review]
Please review my comments when you come back to refactor this work. Good stuff going on 🚀

File: __tests__/app.test.js (hunk 3 above)
Comment (Ben04051) [reply __tests__/app.test.js:45]: Noted, will replace the keys.length check with a test that body is not a key on the object.  Will also make a note to split these out into separate tests at some point and will do that going forward

File: __tests__/app.test.js (hunk 2 above)
Comment (Ben04051) [reply __tests__/app.test.js:51]: Noted, is it sufficient to check that at least 1 element is contained for robustness? i.e. if the test data changes for some reason?

File: __tests__/app.test.js (hunk 1 above)
Comment (Ben04051) [reply __tests__/app.test.js:60]: Noted

**Review by Ben04051 — COMMENTED** [review]
I think this is what you meant? Thanks Poonam!

**Pull request merged**